emain/emain-window.ts

```diff
@@ -699,11 +699,24 @@ ipcMain.on("delete-workspace", (event, workspaceId) => {
             console.log("user cancelled workspace delete", workspaceId, ww?.waveWindowId);
             return;
         }
-        await WorkspaceService.DeleteWorkspace(workspaceId);
-        console.log("delete-workspace done", workspaceId, ww?.waveWindowId);
-        if (ww?.workspaceId == workspaceId) {
-            console.log("delete-workspace closing window", workspaceId, ww?.waveWindowId);
-            ww.destroy();
+        
+        if (ww?.workspaceId == workspaceId){
+            if ( workspaceList?.length > 1 ) {
+                const workspaceIdindex =  workspaceList.findIndex(wse => wse.workspaceid === workspaceId);
+                    
+                const moveIndex = (workspaceIdindex > 0) ? workspaceIdindex - 1 : workspaceIdindex + 1;   
```

Comment:
There's actually a potential for conflict if the next or previous workspace is already claimed by a new window. We need to add a while condition here to iterate through the list until it finds one that isn't claimed by another window. If it can't find it, then it should destroy the window. 

Additionally, I think your change introduces a race between the Go backend and the JS code regarding workspace deletion and workspace switching.

Right now, WorkspaceService.DeleteWorkspace will clean up the window object in the database if there's an attached window. If your call to switchWorkspace is not completed before the backend deletes the workspace (which is likely given the switchWorkspace call is serialized and run in an asynchronous queue), then the backend might clean up the window object, which would result in a blank screen.

This is starting to feel like something we should manage in the Go backend instead of in Electron. A call to WorkspaceService.DeleteWorkspace should determine whether there's another workspace that the window can claim. If not, then it should clean up the window object like it already does and return an empty string to Electron, which should then destroy the window. If it finds a valid workspace to switch to, it should not clean anything up and should return the id for that workspace to Electron, which can then queue a switchWorkspace action.

Sorry for the back-and-forth on this :/